gui/elements/_layer_list.py

```diff
@@ -297,30 +303,61 @@ def _add(self, event):
         """Callback when an item is added to set its order and viewer.
         """
         layer = event.item
-        self._qt.layersList.insert(event.index, len(self), layer)
         layer._order = -len(self)
         layer.viewer = self.viewer
+        layer.events.select.connect(self.viewer._update_active_layers)
+        layer.events.deselect.connect(self.viewer._update_active_layers)
 
     def _remove(self, event):
         """Callback when an item is removed to remove its viewer
         and reset its order.
         """
         layer = event.item
-        self._qt.layersList.remove(layer)
         layer.viewer = None
         layer._order = 0
+        layer.events.select.disconnect(self.viewer._update_active_layers)
+        layer.events.deselect.disconnect(self.viewer._update_active_layers)
 
     def _reorder(self, event):
         """Callback when the list is reordered to propagate those changes
         to the node draw order.
         """
         for i in range(len(self)):
             self[i]._order = -i
-        self._qt.layersList.reorder()
         canvas = self.viewer._canvas
         canvas._draw_order.clear()
         canvas.update()
 
+    def _insert_reorder(self, index, insert):
+        """Reorder list by inserting index and if others selected at insert
+        """
+        total = len(self)
+        indices = [i for i in range(total)]
```

Comment:
This is equivalent to `list(range(total))`

Reply:
And @kne42's form is preferred. ;)